enterprise/server/backends/authdb/authdb.go

```diff
@@ -500,10 +500,9 @@ func (d *AuthDB) LookupUserFromSubID(ctx context.Context, subID string) (*tables
 			LIMIT 1
 		) AS u
 			LEFT JOIN "UserGroups" AS ug
-				ON u.user_id = ug.user_user_id
+				ON u.user_id = ug.user_user_id AND ug.membership_status = ?
```

Comment:
does this worth the same way? I am wondering if the "IS NULL" bit is important for some case where we're creating a new user or something

`AND (ug.membership_status = ? OR ug.user_user_id IS NULL)`

Reply:
@tempoz added this code but I'm not sure if anybody actually remembers the original motivation. 

I think it's very likely at some point the code was 
`WHERE ug.membership_status = ? OR ug.user_user_id IS NULL` which meant that w/o the `ug.user_user_id IS NULL` expression the query would not return any data if the user was not part of any groups. 
In the current code the filter is on the JOIN which means the query WILL still return a row even if the user is not part of any groups.
(Note that there's also a test for a user being no groups...)

Reply:
Pretty sure the code is equivalent - `UserGroups` should never have null user IDs, so the user ID being null is equivalent to 0 rows being matched for the right side of the join.